EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -478,69 +511,81 @@ private static float GetMobilityZScore(PeptideSpectralMatch psm, PeptideWithSetM
             return sequenceToPsmCount;
         }
 
-        //This method ignores ambiguity and loads only the first peptide in a series for each PSM
         public static IEnumerable<PsmData> CreatePsmData(string searchType, List<(string fileName, CommonParameters fileSpecificParameters)> fileSpecificParameters, List<PeptideSpectralMatch> psms, Dictionary<string, int> sequenceToPsmCount, Dictionary<string, Dictionary<int, Tuple<double, double>>> timeDependantHydrophobicityAverageAndDeviation_unmodified, Dictionary<string, Dictionary<int, Tuple<double, double>>> timeDependantHydrophobicityAverageAndDeviation_modified, Dictionary<string, float> fileSpecificMedianFragmentMassErrors, int chargeStateMode, string[] trainingVariables)
         {
             object psmDataListLock = new object();
             List<PsmData> psmDataList = new List<PsmData>();
-
+            List<double> psmOrder = new List<double>();
             int maxThreads = fileSpecificParameters.FirstOrDefault().fileSpecificParameters.MaxThreadsToUsePerFile;
             int[] threads = Enumerable.Range(0, maxThreads).ToArray();
-            Parallel.ForEach(threads, (i) =>
```

Comment:
This approach appears to not be thread safe in my hands. I got crashes with it during calibration task. replaced it with new strategy and no crash